docker/multi_muse/multi_muse_call_p3.py

```diff
@@ -51,20 +52,24 @@ def pcallable():
 
     try:
         output = subprocess.Popen(
-            cmd,
+            shlex.split(cmd),
             shell=shell_var,
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             preexec_fn=child_preexec_set_pdeathsig(),
         )
-        output_stdout, output_stderr = output.communicate()
+        output.wait()
         with lock:
             logger.info("Running command: %s", cmd)
+    except BaseException:
+        output.kill()
+        with lock:
+            logger.error("command failed %s", cmd)
```

Comment:
```python
except BaseException as e:
    output.kill()
    with lock:
        logger.exception(e)
```
Logging the stacktrace could help with debugging too

Reply:
👍